### PR TITLE
Update limiters for loopy

### PR DESCRIPTION
**Pull request by tommbendall**
This updates the code for the `ThetaLimiter` so that it uses loopy kernels (and so will not raise the c parloop warning).

I have also generalised the `ThetaLimiter` so that it will work on the temperature space in 3D.

### CODE REVIEW

**Comment by tommbendall** [comment]
> Looks fine to me. The Jenkin's failure is related to the infamous `h_mountain.py` problem. We will worry about it separately. I think this should be okay. Can you confirm with a short, 3D example that this indeed works?

I can confirm that this limiter does indeed work, I haven't added the example as it would be too long to run in 3D.

**Comment by tommbendall** [comment]
I've just updated the checks for the space being used so that it raises an error if the space used isn't the right one. After this has run the tests then this PR should be ready.

File: gusto/limiters.py [hunk 1]
```diff
@@ -58,63 +23,122 @@ def __init__(self, space):
...
 
         self.Vt = FunctionSpace(space.mesh(), BrokenElement(space.ufl_element()))
-        # check this is the right space, only currently working for 2D extruded mesh
+        # check this is the right space, currently working for 2D and 3D extruded meshes
+        # check that horizontal degree is 1 and vertical degree is 2
+        if self.Vt.ufl_element().degree()[0] != 1 or \
+           self.Vt.ufl_element().degree()[1] != 2:
+            raise ValueError('This is not the right limiter for this space.')
+
         if self.Vt.extruded and self.Vt.mesh().topological_dimension() == 2:
```
Comment (jshipton) [inline gusto/limiters.py]: maybe just have a separate assert that the mesh is extruded

**Comment by jshipton** [comment]
Merging - the only failure is the infamous hydrostatic mountain.

**Pull request merged**